frontend/app/work_packages/controllers/work-package-details-controller.js

```diff
@@ -172,13 +174,53 @@ module.exports = function($scope,
     return !!($scope.workPackage && $scope.workPackage.embedded.watchers !== undefined);
   };
 
-  function displayedActivities(workPackage) {
+  $scope.matchesActivityType = function(activity, type) {
```

Comment:
I'd prefer if we could wire a directive implicitly to a specific resource, but this seems reasonable. I will wrap the check in a helper, although I disagree with the readability issue.